### PR TITLE
Fix a few mismatches between the two link resolver implementations

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: https://github.com/apple/swift-docc/issues/383 rdar://99891308

## Summary

This addresses a couple of mismatches between the documentation cache based link resolver and the path hierarchy based link resolver. There are 3 different types of changes:

- Addressing some false positive mismatches
- Correcting the behavior when running both link resolvers side-by-side at the same time (for comparing the results)
- Fixing a bug in the path hierarchy based resolver where operators weren't always fully disambiguated.

## Dependencies

None

## Testing

Build documentation for a target with many operators (for the operators for a `Comparable` conformance). 

The paths for `<(_:_:)`, `>(_:_:)`, `<=(_:_:)`, and `>=(_:_:)` should have `<` and `>` replaced with `_` and should be disambiguated with hashes. 

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Sources/SwiftDocC/Infrastructure/DocumentationContext.swift [hunk 1]
```diff
@@ -1322,9 +1323,14 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
                     }
                 }
                 
+                if !LinkResolutionMigrationConfiguration.shouldUseHierarchyBasedLinkResolver {
```
Comment (daniel-grumberg) [inline Sources/SwiftDocC/Infrastructure/DocumentationContext.swift:1326]: Should this be merged with the above if block?

**Review by daniel-grumberg — APPROVED** [review]

File: Sources/SwiftDocC/Infrastructure/DocumentationContext.swift (hunk 1 above)
Comment (d-ronnqvist) [reply Sources/SwiftDocC/Infrastructure/DocumentationContext.swift:1326]: Yes. Thanks for noticing.

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**